pages/owned/[ownedPlantId]/index.js

```diff
@@ -72,64 +79,150 @@ export default function DetailsPage() {
       toast.success("Plant removed.");
       router.push("/owned");
     } catch (error) {
-      toast.error(error.message);
+      mutate();
+      toast.error("Failed to delete plant. Please try again.");
     }
   }
 
   return (
     <>
       <StyledHeadline>
         <BackButton href="/owned" />
-        <h2>{plant.nickname || plant.name}</h2>
-        {session && (
-          <Link
-            href={`/owned/${ownedPlantId}/edit`}
-            aria-label="Edit this plant"
-          >
-            <GearIcon size={32} />
-          </Link>
-        )}
+        <StyledHeadlinePlantName>
+          {plant.nickname || plant.name}
```

Comment:
this should already been solved by Dominiks code, plant. name should always be the correct value.